test/installers/knative_installer.go

```diff
@@ -0,0 +1,138 @@
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+)
+
+var (
+	// KnativeEventingYamlClusterInstaller installs Knative eventing cluster wide using YAMLs
+	KnativeEventingYamlClusterInstaller = ServiceInstaller{
+		installClusterWide:   installKnativeEventingUsingYaml,
+		uninstallFromCluster: uninstallKnativeEventingUsingYaml,
+		getAllCrsInNamespace: getKnativeEventingCrsInNamespace,
+	}
+
+	knativeEventingNamespace    = "knative-eventing"
+	knativeEventingVersion      = "v0.20.0"
+	knativeEventingNumberOfPods = 7
+)
+
+// GetKnativeEventingInstaller returns KnativeEventing installer
+func GetKnativeEventingInstaller() *ServiceInstaller {
+	return &KnativeEventingYamlClusterInstaller
+}
+
+func installKnativeEventingUsingYaml() error {
+	// Namespace creation serves as a lock making sure that only first invocation installs Knative eventing, other invocations wait for successful installation
+	if exists, err := framework.CreateNamespaceIfNotExists(knativeEventingNamespace); err != nil {
+		return err
+	} else if exists {
+		framework.GetMainLogger().Info("Detected existing Knative eventing namespace, Knative eventing is supposed to be installed. Checking if related pods are running.")
+		return framework.WaitForPodsInNamespace(knativeEventingNamespace, knativeEventingNumberOfPods, 3)
+	}
+
+	framework.GetMainLogger().Info("Installing Knative eventing")
+
+	output, err := framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/eventing-crds.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing CRDs failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing CRDs")
+	}
+
+	output, err = framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/eventing-core.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing core components failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing core components")
+	}
+
+	output, err = framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/in-memory-channel.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing in-memory channel failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing in-memory channel")
+	}
+
+	output, err = framework.CreateCommand("oc", "apply", "-f", fmt.Sprintf("https://github.com/knative/eventing/releases/download/%s/mt-channel-broker.yaml", knativeEventingVersion)).Execute()
+	if err != nil {
+		framework.GetMainLogger().Error(err, fmt.Sprintf("Applying eventing mt-channel broker failed, output: %s", output))
+		return fmt.Errorf("Error applying eventing mt-channel broker")
+	}
+
+	// Start namespace logging
+	go func() {
```

Comment:
Shouldn't we start the pod log collector before the installation and stop it after the delete of resources ?

Reply:
I will move the logging logic into `service_installer.go` to have it in one place rather than implementing it for every needed service.